core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -487,4 +339,163 @@ public boolean isFinished()
+            return switch (state) {
+                case SET -> Optional.of(convertToDomain(type, blockBuilder.build()));
+                case MIN_MAX -> {
+                    if (minValues == null) {
+                        // all values were null
+                        yield Optional.of(Domain.none(type));
+                    }
+                    Object min = blockToNativeValue(type, minValues);
+                    Object max = blockToNativeValue(type, maxValues);
+                    yield Optional.of(Domain.create(ValueSet.ofRanges(range(type, min, true, max, true)), false));
+                }
+                case NONE -> Optional.empty();
+            };
+        }
+
+        public void disableMinMax()
+        {
+            minMaxPossible = false;
+            if (state == ChannelState.MIN_MAX) {
+                state = ChannelState.NONE;
+                minMaxChannels--;
+            }
+            checkIfEmpty();
+            // Drop references to collected values.
+            minValues = null;
+            maxValues = null;
+        }
+
+        private void updateMinMaxValues(Block block, BlockPositionComparison comparison)
+        {
+            int minValuePosition = -1;
+            int maxValuePosition = -1;
+
+            for (int position = 0; position < block.getPositionCount(); ++position) {
+                if (block.isNull(position)) {
+                    continue;
+                }
+                if (minValuePosition == -1) {
+                    // First non-null value
+                    minValuePosition = position;
+                    maxValuePosition = position;
+                    continue;
+                }
+                if (comparison.compare(block, position, block, minValuePosition) < 0) {
+                    minValuePosition = position;
+                }
+                else if (comparison.compare(block, position, block, maxValuePosition) > 0) {
+                    maxValuePosition = position;
+                }
+            }
+
+            if (minValuePosition == -1) {
+                // all block values are nulls
+                return;
+            }
+            if (minValues == null) {
+                // First Page with non-null value for this block
+                minValues = block.getSingleValueBlock(minValuePosition);
+                maxValues = block.getSingleValueBlock(maxValuePosition);
+                return;
+            }
+            // Compare with min/max values from previous Pages
+            Block currentMin = minValues;
+            Block currentMax = maxValues;
+
+            if (comparison.compare(block, minValuePosition, currentMin, 0) < 0) {
+                minValues = block.getSingleValueBlock(minValuePosition);
+            }
+            if (comparison.compare(block, maxValuePosition, currentMax, 0) > 0) {
+                maxValues = block.getSingleValueBlock(maxValuePosition);
+            }
+        }
+    }
+
+    private void checkIfEmpty()
```

Comment:
Move this below the `isFinished` method

Reply:
It is used by `ChannelFilter` class which is below `isFinished`

Reply:
My expectation is for the private methods of a class to be grouped together and all the private static things to be after the private things.
Right now what we have is private methods, then private static method, then private static class, then private class, then private method. So some reordering is needed for consistency.